lab-trent/lib/clienthandler.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const uuid = require('uuid/v4');
+
+const clients = [];
+
+function Client(conn) {
+  this.nick = 'Guest#' + Math.floor(Math.random() * 99999);
+  this.uuid = uuid();
+  this.ip = conn.remoteAddress;
+  this.socket = conn;
+  conn.on('data', (data) => {
+    this.decode(data);
+  });
+  conn.on('close', () => {
+    clients.splice(clients.indexOf(this));
```

Comment:
You're removing the client from the `pool` but not actually ending the client connection. You'll want to use `.end()` at some point here.